agent/consul/config_replication.go

```diff
@@ -117,12 +118,14 @@ func (s *Server) fetchConfigEntries(lastRemoteIndex uint64) (*structs.IndexedGen
 }
 
 func (s *Server) replicateConfig(ctx context.Context, lastRemoteIndex uint64) (uint64, bool, error) {
+	replicationLogger := s.loggers.Named(logging.Replication).Named(logging.ConfigEntry)
```

Comment:
Doesn't this mean that you're caching the `Replication` logger but not the derived `Replication.ConfigEntry` logger and thus recreating it every replication loop?

Reply:
Yes - but I'm not finding a _cleaner_ way to do this. Any suggestions?

Reply:
in `agent/consul/server.go` make `logger` a derived logger:
```
configReplicatorConfig := ReplicatorConfig{
		Name:     "Config Entry",
		Name:     logging.ConfigEntry,
		Delegate: &FunctionReplicator{ReplicateFn: s.replicateConfig},
		Rate:     s.config.ConfigReplicationRate,
		Burst:    s.config.ConfigReplicationBurst,
		Logger:   logger,
	}
```

in fact for any of those other `ReplicatorConfig` structs you might want to use item-type based subloggers for all of them.

Reply:
Did you want to use this new `logger` function parameter instead of the `replicationLogger` variable now?